lib/spack/spack/cmd/env.py

```diff
@@ -472,6 +473,32 @@ def env_remove(args):
         tty.msg(f"Successfully removed environment '{bad_env_name}'")
 
 
+#
+# env rename
+#
+def env_rename_setup_parser(subparser):
+    """rename an existing environment"""
+    subparser.add_argument(
+        "mv_old_name", metavar="old_name", help="name of existing managed environment to rename"
+    )
+    subparser.add_argument("mv_new_name", metavar="new_name", help="new name for environment")
+
+
+def env_rename(args):
+    """Rename a *named* environment.
```

Comment:
```suggestion
    """Rename a managed environment.
```

Reply:
Support for anonymous environments now included.